Adddefaults

blainehowerton opened the pull request:
Jeff,

I added a new model and table called defaults.  This has the default values for retailer_split, cause_split, and a new fee_split (dig in's fee).  The transaction.new controller references this default table to save the transaction fee percentage from the new defaults table.  This allows us to change that defaults for these values without having to do a migration or going into the code itself.

in this branch, I also compiled the database state into a migration file called create structure.  I was having some problems with the migrations and I wanted to start fresh on the migration side.

This all will require you to re-load your local database (rake:db:reset), and then call (rake:db:seed) to get all of your initial data set.  I entered a few more seeds to get the database in a usable state from the get go.  This will also probably help you get back to being able to use the typeahead (by resetting and re-seeding whatever happened the last time).

Let me know if you have any problems getting your local database back up and running.  I'll probably merge this branch tonight and begin work on planning for the new module for deposits.

Commit:
added defaults model and table, and revised new transaction controller to save fee split from the new default table

Commit:
created default values and implemented into transaction new view, put defaults and other seed information into database for future resets.

Commit:
deleted previous migration files.

Commit:
compiled previous migration values into a create structure file, in order to get rid of old migration files

Ninjef commented:
Module for deposits? Were we gonna have three new tables added? Or are you
talking about something else?

On Wed, Mar 30, 2016, 13:32 blainehowerton notifications@github.com wrote:

> Jeff,
> 
> I added a new model and table called defaults. This has the default values
> for retailer_split, cause_split, and a new fee_split (dig in's fee). The
> transaction.new controller references this default table to save the
> transaction fee percentage from the new defaults table. This allows us to
> change that defaults for these values without having to do a migration or
> going into the code itself.
> 
> in this branch, I also compiled the database state into a migration file
> called create structure. I was having some problems with the migrations and
> I wanted to start fresh on the migration side.
> 
> This all will require you to re-load your local database (rake:db:reset),
> and then call (rake:db:seed) to get all of your initial data set. I entered
> a few more seeds to get the database in a usable state from the get go.
> This will also probably help you get back to being able to use the
> typeahead (by resetting and re-seeding whatever happened the last time).
> 
> Let me know if you have any problems getting your local database back up
> and running. I'll probably merge this branch tonight and begin work on
> 
> ## planning for the new module for deposits.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/blainehowerton/diginrewards/pull/11
> Commit Summary
> - added defaults model and table, and revised new transaction
>   controller to save fee split from the new default table
> - created default values and implemented into transaction new view,
>   put defaults and other seed information into database for future resets.
> - deleted previous migration files.
> - compiled previous migration values into a create structure file, in
>   order to get rid of old migration files
> 
> File Changes
> - _M_ app/controllers/application_controller.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-0
>   (2)
> - _M_ app/controllers/transactions_controller.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-1
>   (11)
> - _A_ app/models/default.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-2
>   (2)
> - _M_ app/models/transaction.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-3
>   (7)
> - _M_ app/views/causes/new.html.erb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-4
>   (4)
> - _M_ config/initializers/rails_admin.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-5
>   (2)
> - _D_ db/migrate/20160215215313_create_users.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-6
>   (10)
> - _D_ db/migrate/20160215215352_create_causes.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-7
>   (10)
> - _D_ db/migrate/20160215215430_create_retailers.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-8
>   (9)
> - _D_ db/migrate/20160215215659_create_transactions.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-9
>   (11)
> - _D_ db/migrate/20160218172939_add_fields_to_tables.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-10
>   (36)
> - _D_ db/migrate/20160221200715_remove_balances.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-11
>   (14)
> - _D_ db/migrate/20160229045303_add_image_to_projects.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-12
>   (7)
> - _D_ db/migrate/20160315032204_remove_email_from_users.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-13
>   (4)
> - _D_ db/migrate/20160318024822_add_devise_to_users.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-14
>   (49)
> - _D_ db/migrate/20160321023225_devise_create_admins.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-15
>   (21)
> - _A_ db/migrate/20160330190101_create_structure.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-16
>   (95)
> - _M_ db/schema.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-17
>   (38)
> - _A_ db/schema_backup
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-18
>   (111)
> - _M_ db/seeds.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-19
>   (5)
> - _A_ test/fixtures/defaults.yml
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-20
>   (11)
> - _A_ test/models/default_test.rb
>   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-21
>   (7)
> 
> Patch Links:
> - https://github.com/blainehowerton/diginrewards/pull/11.patch
> - https://github.com/blainehowerton/diginrewards/pull/11.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/blainehowerton/diginrewards/pull/11

blainehowerton commented:
Sorry - I meant the new tables.
On Wed, Mar 30, 2016 at 4:48 PM Ninjef notifications@github.com wrote:

> Module for deposits? Were we gonna have three new tables added? Or are you
> talking about something else?
> 
> On Wed, Mar 30, 2016, 13:32 blainehowerton notifications@github.com
> wrote:
> 
> > Jeff,
> > 
> > I added a new model and table called defaults. This has the default
> > values
> > for retailer_split, cause_split, and a new fee_split (dig in's fee). The
> > transaction.new controller references this default table to save the
> > transaction fee percentage from the new defaults table. This allows us to
> > change that defaults for these values without having to do a migration or
> > going into the code itself.
> > 
> > in this branch, I also compiled the database state into a migration file
> > called create structure. I was having some problems with the migrations
> > and
> > I wanted to start fresh on the migration side.
> > 
> > This all will require you to re-load your local database (rake:db:reset),
> > and then call (rake:db:seed) to get all of your initial data set. I
> > entered
> > a few more seeds to get the database in a usable state from the get go.
> > This will also probably help you get back to being able to use the
> > typeahead (by resetting and re-seeding whatever happened the last time).
> > 
> > Let me know if you have any problems getting your local database back up
> > and running. I'll probably merge this branch tonight and begin work on
> > 
> > ## planning for the new module for deposits.
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> > https://github.com/blainehowerton/diginrewards/pull/11
> > Commit Summary
> > - added defaults model and table, and revised new transaction
> >   controller to save fee split from the new default table
> > - created default values and implemented into transaction new view,
> >   put defaults and other seed information into database for future resets.
> > - deleted previous migration files.
> > - compiled previous migration values into a create structure file, in
> >   order to get rid of old migration files
> > 
> > File Changes
> > - _M_ app/controllers/application_controller.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-0
> >   (2)
> > - _M_ app/controllers/transactions_controller.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-1
> >   (11)
> > - _A_ app/models/default.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-2
> >   (2)
> > - _M_ app/models/transaction.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-3
> >   (7)
> > - _M_ app/views/causes/new.html.erb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-4
> >   (4)
> > - _M_ config/initializers/rails_admin.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-5
> >   (2)
> > - _D_ db/migrate/20160215215313_create_users.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-6
> >   (10)
> > - _D_ db/migrate/20160215215352_create_causes.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-7
> >   (10)
> > - _D_ db/migrate/20160215215430_create_retailers.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-8
> >   (9)
> > - _D_ db/migrate/20160215215659_create_transactions.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-9
> >   (11)
> > - _D_ db/migrate/20160218172939_add_fields_to_tables.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-10
> >   (36)
> > - _D_ db/migrate/20160221200715_remove_balances.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-11
> >   (14)
> > - _D_ db/migrate/20160229045303_add_image_to_projects.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-12
> >   (7)
> > - _D_ db/migrate/20160315032204_remove_email_from_users.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-13
> >   (4)
> > - _D_ db/migrate/20160318024822_add_devise_to_users.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-14
> >   (49)
> > - _D_ db/migrate/20160321023225_devise_create_admins.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-15
> >   (21)
> > - _A_ db/migrate/20160330190101_create_structure.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-16
> >   (95)
> > - _M_ db/schema.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-17
> >   (38)
> > - _A_ db/schema_backup
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-18
> >   (111)
> > - _M_ db/seeds.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-19
> >   (5)
> > - _A_ test/fixtures/defaults.yml
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-20
> >   (11)
> > - _A_ test/models/default_test.rb
> >   https://github.com/blainehowerton/diginrewards/pull/11/files#diff-21
> >   (7)
> > 
> > Patch Links:
> > - https://github.com/blainehowerton/diginrewards/pull/11.patch
> > - https://github.com/blainehowerton/diginrewards/pull/11.diff
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly or view it on GitHub
> > https://github.com/blainehowerton/diginrewards/pull/11
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/blainehowerton/diginrewards/pull/11#issuecomment-203671962
> 
> ## 
> 
> Blaine Howerton
> Denver Media Center
> Yellow Gal High Definition
> 720.255.1640 xt 101

Commit:
added footer menu to onboarding and transaction views
